src/js/components/DataTable/Body.js

```diff
@@ -264,29 +324,50 @@ const Body = forwardRef(
                 if (typeof onClickRow === 'function') {
                   event.persist();
                   const adjustedEvent = event;
-                  adjustedEvent.datum = data?.[active];
+                  adjustedEvent.datum = data?.[focused];
                   onClickRow(adjustedEvent);
                 } else if (onClickRow === 'select') {
                   selectRow();
                 }
               }
             : undefined
         }
-        onUp={onClickRow && active ? () => setActive(active - 1) : undefined}
+        onUp={
```

Comment:
If focused is 0 (so the first row is focused), pressing the up arrow will set previousIndex to -1?
I wasnt sure about this but I did check locally if I was on the first row and pressed up arrow it stays on the first row but logic wise just wanted to make sure we don't get unintended behavior

Reply:
ah okay thank you for explanation